lib/ansiblelint/runner.py

```diff
@@ -77,29 +77,37 @@ def is_excluded(self, file_path: str) -> bool:
     def run(self) -> List[MatchError]:
         """Execute the linting process."""
         files: List[Lintable] = list()
+        matches: List[MatchError] = list()
+
         for playbook in self.playbooks:
             if self.is_excluded(str(playbook.path.resolve())) or playbook.kind == 'role':
                 continue
             files.append(playbook)
-        matches = set(self._emit_matches(files))
-
-        # remove duplicates from files list
-        files = [value for n, value in enumerate(files) if value not in files[:n]]
-
-        # remove files that have already been checked
-        files = [x for x in files if x.path not in self.checked_files]
-        for file in files:
-            _logger.debug(
-                "Examining %s of type %s",
-                ansiblelint.file_utils.normpath(file.path),
-                file.kind)
-            matches = matches.union(
-                self.rules.run(file, tags=set(self.tags),
-                               skip_list=self.skip_list))
+            matches.extend(AnsibleSyntaxCheckRule._get_ansible_syntax_check_matches(playbook))
+
+        if not matches:  # do our processing only when ansible syntax check passed
+
+            matches.extend(self._emit_matches(files))
+
+            # remove duplicates from files list
+            files = [value for n, value in enumerate(files) if value not in files[:n]]
+
+            # remove files that have already been checked
+            files = [x for x in files if x.path not in self.checked_files]
+            for file in files:
+                _logger.debug(
+                    "Examining %s of type %s",
+                    ansiblelint.file_utils.normpath(file.path),
+                    file.kind)
+
+                matches.extend(
+                    self.rules.run(file, tags=set(self.tags),
+                                   skip_list=self.skip_list))
+
         # update list of checked files
         self.checked_files.update([str(x.path) for x in files])
 
-        return sorted(matches)
+        return sorted(list(set(matches)))
```

Comment:
Nitpick: `sorted` know how to handle sets (or any other iterabel), so no need to convert to list before sorting.

Reply:
True, i will fix it in follow-up, I am sure that I will touch the same method very soon, before the release.